tests/cpp-tests/Classes/FileUtilsTest/FileUtilsTest.cpp

```diff
@@ -592,6 +593,127 @@ std::string TestWriteString::subtitle() const
     return "";
 }
 
+class CustomBuffer : public ResizableBuffer {};
+
+struct AlreadyExistsBuffer {};
+
+NS_CC_BEGIN
```

Comment:
But i think it is not explicit specialization here, it is inheritance here. Did you met compiling error? It works ok when when i

* remove NS_CC_BEGIN NS_CC_END
* and changed to `class ResizableBufferAdapter<AlreadyExistsBuffer> : public cocos2d::ResizableBuffer`